[quant][graphmodel] linear_relu

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#40021 [quant][graphmodel] linear_relu**

This replaces #36889 due to significant merge conflicts

Differential Revision: [D22087061](https://our.internmc.facebook.com/intern/diff/D22087061)

Commit:
[quant][graphmodel] linear_relu

This replaces #36889 due to significant merge conflicts

[ghstack-poisoned]

Commit:
Update on "[quant][graphmodel] linear_relu"


This replaces #36889 due to significant merge conflicts

[ghstack-poisoned]

Commit:
Update on "[quant][graphmodel] linear_relu"


This replaces #36889 due to significant merge conflicts

[ghstack-poisoned]

jerryzh168 commented on test/quantization/test_quantize_script.py:
```diff
@@ -1233,6 +1233,98 @@ def _test_op_impl(self, module, data, quantized_op, tracing=False, debug=False,
         return models[False]
 
+    @skipIfNoFBGEMM
+    def test_quantized_linear(self):
```
could you merge all the tests in one function? and add tests for tracing as well?

z-a-f replied:
Is there an example of a tracing tests that I could base on?

jerryzh168 replied:
for example: https://github.com/pytorch/pytorch/blob/master/test/quantization/test_quantize_script.py#L1237

Commit:
Update on "[quant][graphmodel] linear_relu"


This replaces #36889 due to significant merge conflicts

[ghstack-poisoned]

Commit:
Update on "[quant][graphmodel] linear_relu"


This replaces #36889 due to significant merge conflicts

[ghstack-poisoned]

Commit:
Update on "[quant][graphmodel] linear_relu"


This replaces #36889 due to significant merge conflicts

[ghstack-poisoned]

jerryzh168 commented on test/quantization/test_quantize_script.py:
```diff
@@ -1233,6 +1233,60 @@ def _test_op_impl(self, module, data, quantized_op, tracing=False, debug=False,
+                        self.relu = torch.nn.ReLU()
+                else:
+                    self.relu = torch.nn.Identity()
+            def forward(self, x):
```
lint

jerryzh168 commented:
CI is broken, please take a look

jerryzh168 reviewed (approved):
code looks good, please fix lint and CI before landing.

z-a-f commented:
> CI is broken, please take a look

I am not sure why the CI is breaking -- it looks like for some reason unrelated tests get triggered

Commit:
Update on "[quant][graphmodel] linear_relu"


This replaces #36889 due to significant merge conflicts

Differential Revision: [D22087061](https://our.internmc.facebook.com/intern/diff/D22087061)

[ghstack-poisoned]

Commit:
Update on "[quant][graphmodel] linear_relu"


This replaces #36889 due to significant merge conflicts

Differential Revision: [D22087061](https://our.internmc.facebook.com/intern/diff/D22087061)

[ghstack-poisoned]

Commit:
Update on "[quant][graphmodel] linear_relu"


This replaces #36889 due to significant merge conflicts

Differential Revision: [D22087061](https://our.internmc.facebook.com/intern/diff/D22087061)

[ghstack-poisoned]

Commit:
Update on "[quant][graphmodel] linear_relu"


This replaces #36889 due to significant merge conflicts

Differential Revision: [D22087061](https://our.internmc.facebook.com/intern/diff/D22087061)

[ghstack-poisoned]

Commit:
Update on "[quant][graphmodel] linear_relu"


This replaces #36889 due to significant merge conflicts

Differential Revision: [D22087061](https://our.internmc.facebook.com/intern/diff/D22087061)

[ghstack-poisoned]

Commit:
Update on "[quant][graphmodel] linear_relu"


This replaces #36889 due to significant merge conflicts

Differential Revision: [D22087061](https://our.internmc.facebook.com/intern/diff/D22087061)

[ghstack-poisoned]
